crates/loader/src/local/mod.rs

```diff
@@ -93,6 +98,37 @@ fn error_on_duplicate_ids(components: Vec<RawComponentManifest>) -> Result<()> {
     Ok(())
 }
 
+fn validate_raw_app_manifest(raw: &RawAppManifestAnyVersion) -> Result<()> {
+    match raw {
+        RawAppManifestAnyVersion::V1(raw) => {
```

Comment:
I'm also wondering if we should set up more general validation infrastructure..

Reply:
I agree that this would not be the best place to put validator, as each new version would require some copy and paste of previous version's validations. Building a general validation infras would be hard, but I like the idea. It needs to be versioned too.

Reply:
Hi @itowlson and @Mossaka, thanks for the suggestion. Currently, if we want to set validation on the `Application` object, only `up` and `trigger` commands will be validated. Other commands like `build` or `deploy` use `RawAppManifest` directly. For validating `spin.toml` before runtime, I think adding validation in the `Application` object is enough. If you all agree with this, I can update this PR later.

Reply:
`build` shouldn't care about configuration validity, unless used with `--up`, and I'd expect it that part to go through `Application`.  Good point about `deploy` (and `publish`) though, we'd want to catch it there and they don't build `Application` abstractions.

Given that, the best course is, I think, to go ahead with the validation where you have it, and raise an issue to revisit the consolidation issue as part of a general validation infrastructure, which might require more invasive changes.  @Mossaka does that make sense to you?